python/semantic_kernel/connectors/memory/qdrant/qdrant_memory_store.py

```diff
@@ -0,0 +1,385 @@
+from semantic_kernel.memory.memory_store_base import MemoryStoreBase
+from semantic_kernel.utils.null_logger import NullLogger
+
+
+class QdrantMemoryStore(MemoryStoreBase):
+    _qdrantclient: QdrantClient
+    _logger: Logger
+
+    def __init__(self, hostip: str, port: Optional[int] = 6333, logger: Optional[Logger] = None) -> None:
+        self._qdrantclient = QdrantClient(host=hostip, port=port)
+        self._logger = logger or NullLogger()
+
+        """Initializes a new instance of the QdrantMemoryStore class.
+
+        Arguments:
+            logger {Optional[Logger]} -- The logger to use. (default: {None})
+        """
+
+
+    async def create_collection_async(self, collection_name: str, vector_size: Optional[int] = 1536, distance: Optional[str]="Cosine") -> None:
+        """Creates a new collection if it does not exist.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to create.
+            vector_size {Optional[int]} -- The size of the vector. (default: {1536})
+            distance {Optional[str]} -- The distance metric to use. (default: {"Cosine"})
+        Returns:
+            None
+        """
+
+        self._qdrantclient.recreate_collection(
+            collection_name="{collection_name}",
+            vectors_config=VectorParams(size=vector_size, distance=Distance.COSINE),
+            )
+    
+    
+    async def get_collections_async(
+        self,
+    ) -> List[str]:
+        """Gets the list of collections.
+
+        Returns:
+            List[str] -- The list of collections.
+        """
+        return list(self._qdrantclient.get_collections())
+    
+
+    async def delete_collection_async(self, collection_name: str) -> None:
+        """Deletes a collection.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to delete.
+
+        Returns:
+            None
+        """
+
+        self._qdrantclient.delete_collection(collection_name=collection_name)
+
+
+    async def does_collection_exist_async(self, collection_name: str) -> bool:
+        """Checks if a collection exists.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to check.
+
+        Returns:
+            bool -- True if the collection exists; otherwise, False.
+        """
+
+        collection_info = self._qdrantclient.get_collection(collection_name=collection_name)
+
+        if collection_info.status == CollectionStatus.GREEN:
+            return collection_name
+        else:
+            return ""
+
+
+    async def upsert_async(self, collection_name: str, record: MemoryRecord) -> str:
+        """Upserts a record.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to upsert the record into.
+            record {MemoryRecord} -- The record to upsert.
+
+        Returns:
+            str -- The unqiue database key of the record.
+        """
+        record._key = record._id
```

Comment:
see the .NET implementation - the ID passed via a MemoryRecord will most likely be a natural language string. For QdrantMemoryStore, we need to distinguish between point IDs and the user-provided key (which ends up in the payload)

Reply:
Created conversion function emulating .NET implementation in commit in qdrant_utils.py: https://github.com/microsoft/semantic-kernel/pull/919/commits/a0732ee615e58792189d17ed7e192908f61c7eee

please review.